### ISSUE TITLE
Move helm chart

### ISSUE THREAD

**Issue by onedr0p** [bitnami-labs/sealed-secrets#389]
> TL;DR the sealed-secrets project is going to maintain the helm chart (one of the key motivator is to ensure it will be kept backwards compatible and stay in sync with the main release manifest)

---

Seeing as the official Helm chart repo will be deprecated on Nov 13th 2020, does it make sense to maintain this chart here instead?

https://github.com/helm/charts/tree/master/stable/sealed-secrets

**Comment by mkmik** [comment]
The helm chart is maintained by the community; the sealed-secrets maintainers currently don't want to maintain the helm chart. If the current maintainers want to keep maintaining it they should move it somewhere.

Otherwise I can ask somebody at bitnami to maintain the helm chart for sealed secrets but last time I checked that involved putting the helm chart in a new git repo anyway (due to the way the application catalog team releases and tests helm chart, which includes lots of automation which mandates some uniformity)

**Comment by onedr0p** [comment]
> Otherwise I can ask somebody at bitnami to maintain the helm chart for sealed secrets but last time I checked that involved putting the helm chart in a new git repo anyway (due to the way the application catalog team releases and tests helm chart, which includes lots of automation which mandates some uniformity)

Agreed, the bitnami team has it's own repo for helm charts, this makes sense to move it here.

https://github.com/bitnami/charts

**Comment by onedr0p** [comment]
Any update on this? We're getting closer and closer to the official helm charts being deprecated.

**Comment by mkmik** [comment]
Have you reached out to the current maintainer?

We cannot move the helm chart to the bitnami repo without effective forking the chart.

**Comment by davidkarlsen** [comment]
@carrodher If you decide on moving it to the bitnami chart repo I hope you can make the initial version compatible with the one in stable, as having to reinstall will mean problems due to keys disappearing etc

**Comment by mkmik** [comment]
Hi, I'm the sealed-secrets project maintainer, please read https://github.com/bitnami/charts/pull/4017#issuecomment-713913133

TL;DR the sealed-secrets project is going to maintain the helm chart (one of the key motivator is to ensure it will be kept backwards compatible and stay in sync with the main release manifest)

**Comment by mkmik** [comment]
I will keep this task open until the chart is actually moved here

**Comment by JohnLBevan** [comment]
FYI: There's been [an announcement](https://helm.sh/blog/new-location-stable-incubator-charts/) on the official `stable` and `incubator` repos for charts: 

- https://charts.helm.sh/stable
- https://charts.helm.sh/incubator

I'm not sure whether this influences the above discussion?

**Comment by mkmik** [comment]
not really, the location problem just brought this topic to our attention; I believe consolidating the chart in the main repository will be beneficial to the community

**Comment by alexellis** [comment]
@stefanprodan - what are Weaveworks going to be doing - where you consume SealedSecrets for Flux etc?

**Comment by alexellis** [comment]
This needs to be pinned somewhere ->

https://github.com/bitnami-labs/sealed-secrets/issues/389#issuecomment-717863252

**Comment by mkmik** [comment]
Updated the issue description and mentioned the decision to onboard the chart

**Comment by adusumillipraveen** [comment]
Has there been any update on this move as there is only a couple of days left for stable to be deprecated.

**Comment by mkmik** [comment]
I'm working on it; there were some yaks to be shaven (e.g. migrating off travis ci because of reasons)

**Comment by fliphess** [comment]
@mkmik 

I'm in the middle of a migration from helm2 to helm3 and a newer k8s version so I could use the new chart location.  

Is there anything I can do to help you get there? 

If you need testers, please highlight me, I'm available :)
(No pressure btw, just offering help)

**Comment by JohnLBevan** [comment]
For anyone waiting for this migration, a temporary solution is to use the new stable; i.e.

```
spec:
  releaseName: sealed-secrets
  chart:
    #repository: https://kubernetes-charts.storage.googleapis.com/ # FROM THIS
    repository: https://charts.helm.sh/stable # TO THIS
    name: sealed-secrets
    version: 1.10.0
```

Longer term it makes sense to switch to Bitnami's solution once they get it up; but the above buys you time.

**Comment by davidkarlsen** [comment]
@mkmik Do you need any help on this?

**Comment by mkmik** [comment]
Yes please. I need:

1. A PR that copies the latest chart into the "/helm" dir in this repo.

2. a GH actions workflow that builds the chart and pushes it into a OCI registry. Since running that requires a token, I think I'll need to do the last step myself, but it would help having a working example in some demo repo; help appreciated!

**Comment by fliphess** [comment]
Working on a PR! :)

**Comment by stefanprodan** [comment]
> a GH actions workflow that builds the chart and pushes it into a OCI registry

Can we please use GH Pages instead of OCI? OCI support is alpha in Helm and it will not work with Flux. 

Here is an action that publishes charts to GH Pages it's being used for Flux, Flagger, OpenServiceMesh, SMI and other CNCF projects. https://github.com/stefanprodan/helm-gh-pages

**Comment by fliphess** [comment]
I've created an initial PR here: https://github.com/bitnami-labs/sealed-secrets/pull/483

Open to all suggestions as not all information is correct like owners etc.

**Comment by alexellis** [comment]
+1 to GitHub pages, see also `make charts` in [inlets-operator](https://github.com/inlets/inlets-operator) for a simple example to copy.

**Comment by fliphess** [comment]
I'm a bit busy at work atm due to some sudden changes in planning... If there's anyone willing to create a second PR or add the gh-pages to mine that would be awesome. 
Otherwise I can do it later this week / after work.

**Comment by mkmik** [comment]
Ok for GH-pages.

**Comment by davidkarlsen** [comment]
Started an attempt at https://github.com/bitnami-labs/sealed-secrets/runs/1516963120?check_suite_focus=true#step:9:25 - but the resources are a bit out of date (old RBACs, old crd spec). Do you have an updated CRD?

**Comment by mkmik** [comment]
habemus chartam gubernāculī 🥳 🎉 🎈 

```
$ helm repo add sealed-secrets https://bitnami-labs.github.io/sealed-secrets
$ helm install sealed-secrets/sealed-secrets
```

Sorry for taking so much time, despite the community being so willing to help I was the choke-point.

Please try it out. I'm interested here only in regressions from the previous helm chart. Other improvements will be handled in other tickets and from now on, changes in the manifests in the Plain old YAML release and the helm chart will be in sync.

**Comment by mkmik** [comment]
Readme updated. Closing
